src/Handlers/AbstractPaymentMethodHandler.php

```diff
@@ -870,7 +870,7 @@ public function handleAdyenOrderPayment(
         ) - $partialAmount;
 
         // Remove the used state.data
-        $this->paymentStateDataService->deletePaymentStateDataFromContextToken($salesChannelContext->getToken());
+        $this->paymentStateDataService->deletePaymentStateData($storedStateData['id']);
```

Comment:
When the $storedStateData is null, there can be an issue.

Reply:
For here, we already check the null status of the $storedStateData